src/reachy2_sdk/reachy_sdk.py

```diff
@@ -324,51 +313,17 @@ def _setup_parts(self) -> None:
             self._mobile_base = MobileBase(self._robot.head, initial_state.mobile_base_state, self._grpc_channel)
             self.info._set_mobile_base(self._mobile_base)
 
-    async def _wait_for_stop(self) -> None:
-        while not self._stop_flag.is_set():
-            await asyncio.sleep(0.1)
-        if self._lost_connection:
-            raise ConnectionError("Connection with Reachy lost, check the sdk server status.")
-
     def _start_sync_in_bg(self) -> None:
```

Comment:
This function may be not useful anymore, we can directly call _get_stream_update_loop from the thread.start, and open the channel from within.